include/boost/url/detail/any_allocator.hpp

```diff
@@ -0,0 +1,175 @@
+//
+// Copyright (c) 2022 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/url
+//
+
+
+#ifndef BOOST_URL_DETAIL_ANY_ALLOCATOR_HPP
+#define BOOST_URL_DETAIL_ANY_ALLOCATOR_HPP
+
+#include <boost/url/detail/type_id.hpp>
+
+#include <boost/core/allocator_access.hpp>
+#include <boost/core/empty_value.hpp>
+#include <boost/type_traits/type_with_alignment.hpp>
+
+#include <cstddef>
+
+namespace boost {
+namespace urls {
+namespace detail {
+
+struct any_allocator_base
+{
+    struct base {
```

Comment:
Why do these braces keep showing up on the same line as the preceding text?

Reply:
Because I have to do everything manually to replicate the formatting you like and (unlike clang-format) I fail sometimes.

Mimicking the style very often takes much more time than coding. I've been working so hard on style that I even wrote a small program just to try to infer the formatting you like from the source code: https://github.com/alandefreitas/clang-unformat 

This helped a little but I still can't apply it to the whole file because some patterns, such as 

```cpp
template <class T>
any_allocator<T>::
any_allocator() noexcept
: any_allocator(std::allocator<char>{})
{
}
```

are very peculiar and have no correspondence in clang-format, which would give me:

```cpp
template<class T>
any_allocator<T>::any_allocator() noexcept
    : any_allocator(std::allocator<char>{})
{
}
```

so I have to keep doing that manually and errors are going to happen eventually.

Reply:
It seems like the problem is that your editor insists on putting the brace on the same line. This is not just a "formatting issue" - in order to set a breakpoint after the arguments are constructed but before a function executes, the brace needs to be on its own line. And this also affects coverage.

Reply:
> It seems like the problem is that your editor insists on putting the brace on the same line. 

The editor does nothing. It's all manual. If I let the editor do its thing, it's going to remove the formatting you like in other places, I'm not going to notice, and then you're going to complain about that too, the IDE will remove formatting at some other place, and we are never going to move forward.

> This is not just a "formatting issue" - in order to set a breakpoint after the arguments are constructed but before a function executes, the brace needs to be on its own line. And this also affects coverage.

I agree 100% with the rationale for breaking before braces. You're correct and the IDE is set to do that.  

The thing is breaking before braces is not the problem at all. For instance, in

```cpp
template <class T>
any_allocator<T>::
any_allocator() noexcept
: any_allocator(std::allocator<char>{})
{
}
```

why would you need to set a breakpoint at

```cpp
any_allocator<T>::
```

?

The issue is not breaking before braces here. Braking before braces is very simple, and the editor is set to do that even without clang-format or anything. Even notepad can probably do that. 

The real problem is I have to turn off any kind of automatic formatting, including breaking before braces, to maintain other conventions that have nothing to with that. Even with clang-format, which has hundreds of options, there is no option that produces:

```cpp
any_allocator<T>::
any_allocator() noexcept
```

At this point, my `.clang-format` matches your style almost perfectly, but I have to keep it off so it doesn't break these small conventions. I only apply it 

For instance, if I turn it on and apply it completely to `impl/any_allocator.hpp`, I get [this](https://gist.github.com/alandefreitas/abf0004ca224b38459488f7dedd9c495), which matches the functional components of your style completely with no room for mistakes. But it's going to break the conventions such as 

```
any_allocator<T>::
```

or 

```
keywords
return_type
type::
```

so I can't keep applying it to everything I commit. 

When things are manual, there are going to be mistakes here and there. I can fix them. No problem. But they are going to happen every once in a while.